src/post/post-interaction-counts.service.unit.test.ts

```diff
@@ -0,0 +1,278 @@
+import type { Logger } from '@logtape/logtape';
+import { error, ok } from 'core/result';
+import { beforeEach, describe, expect, it, vi } from 'vitest';
+import { PostInteractionCountsService } from './post-interaction-counts.service';
+import type { Post } from './post.entity';
+import type { KnexPostRepository } from './post.repository.knex';
+import type { PostService } from './post.service';
+
+describe('PostInteractionCountsService', () => {
+    let service: PostInteractionCountsService;
+    let mockPostService: PostService;
+    let mockPostRepository: KnexPostRepository;
+    let mockLogger: Logger;
+
+    beforeEach(() => {
+        mockPostService = {
+            updateInteractionCounts: vi.fn(),
+        } as unknown as PostService;
+        mockPostRepository = {
+            getById: vi.fn(),
+        } as unknown as KnexPostRepository;
+        mockLogger = {
+            info: vi.fn(),
+            error: vi.fn(),
+        } as unknown as Logger;
+        service = new PostInteractionCountsService(
+            mockPostService,
+            mockPostRepository,
+            mockLogger,
+        );
+    });
+
+    describe('updateInteractionCounts', () => {
+        it('skips updating interaction counts if post is not found and logs an error', async () => {
+            const postId = 999;
+            vi.mocked(mockPostRepository.getById).mockResolvedValue(null);
+
+            await service.updateInteractionCounts([postId]);
+
+            expect(
+                mockPostService.updateInteractionCounts,
+            ).not.toHaveBeenCalled();
+            expect(mockLogger.error).toHaveBeenCalledWith(
+                'Post with ID {postId} not found when updating interaction counts - Skipping',
+                { postId },
+            );
+        });
+
+        it('logs an error if updating interaction counts fails', async () => {
+            const postId = 1;
+            const publishedAt = new Date(Date.now() - 2 * 60 * 60 * 1000);
```

Comment:
It might be good to put these time values into consts to make it easier to read / re-use

```ts
const TWO_HOURS = 2 * 60 * 60 * 1000;
```

Reply:
Good one, improved readability in d91d5ecf3b0a76f5a712d5bbc4e88523ac284d5f